common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -570,6 +600,79 @@ class NetworkSensorManager : SensorManager {
         })
     }
 
+    @SuppressLint("MissingPermission")
+    private fun updateCellConnectionSensor(context: Context) {
+        if (!isEnabled(context, cellConnection) || !hasTelephony(context)) {
+            return
+        }
+
+        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+
+        val cell = telephonyManager.getAllCellInfo().findLast { cell -> cell.isRegistered } ?: return
+
+        val cellId : String = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            when (cell) {
+                is CellInfoNr -> {
+                    "nr:" + (cell.cellIdentity as CellIdentityNr).nci
+                }
+
+                is CellInfoTdscdma -> {
+                    "tdcdma:" + cell.cellIdentity.lac + ":" + cell.cellIdentity.cid
+                }
+
+                is CellInfoLte -> { //tested
+                    "lte:" + cell.cellIdentity.tac + ":" + cell.cellIdentity.ci + ":" + cell.cellIdentity.pci
+                }
+
+                is CellInfoWcdma -> { //tested
+                    "wcdma:" + cell.cellIdentity.lac + ":" + cell.cellIdentity.cid
+                }
+
+                is CellInfoCdma -> {
+                    "cdma:" + cell.cellIdentity.systemId + ":" + cell.cellIdentity.networkId + ":" + cell.cellIdentity.basestationId
```

Comment:
would providing latitude and longitude make more sense for this case? noticed that this particular `cellIdentity` provides it

https://developer.android.com/reference/kotlin/android/telephony/CellIdentityCdma

too bad none of the other do :)

Reply:
Indeed, I missed it. It would make things so simple if all of them supported it ...
I can change it, but I will not be able to test it :sweat_smile:

Reply:
maybe we try and keep the state more predictable so its usable, we can always add things to the attributes that describe the state. 

I noticed there seems to be a database of cell tower location that can be used with the cell tower identity, pretty neat!